src/integrations/store/actions/index.js

```diff
@@ -24,6 +25,15 @@ export const logoutProvider = () => async (dispatch, getState) => {
 }
 
 export const runProviderUpdate = (provider, data) => async (dispatch, getState) => {
+  if (provider.constructor.providerName === WALLET_PROVIDER.METAMASK && isFeatureEnabled('tournament')) {
```

Comment:
@mmvsha Hmm, so when the user uses uPort, the address will remain?

Reply:
I think this part can be removed, it should always delete the mainnet address when the wallet is locked/closed

Reply:
we don't set mainnet address for uPort users @andre-meyer 
how do you detect if the wallet is closed?

Reply:
This was resolved :) Feel free to merge